server/schedule/basic_cluster.go

```diff
@@ -41,15 +41,49 @@ const (
 	hotRegionAntiCount           = 1
 )
 
-// BasicCluster provides basic data member and interface for a tikv cluster
+// BasicCluster provides basic data member and interface for a tikv cluster.
 type BasicCluster struct {
 	Stores          *core.StoresInfo
 	Regions         *core.RegionsInfo
 	WriteStatistics cache.Cache
 	ReadStatistics  cache.Cache
 }
 
-// NewBasicCluster creates a BasicCluster
+// NewDiffMap creates a DiffMap.
+func NewDiffMap(operators []*Operator, cluster Cluster) DiffMap {
+	m := make(map[uint64]*StoreDiff)
+
+	for _, op := range operators {
+		if !op.IsTimeout() && !op.IsFinish() {
+			op.Influence(m, cluster.GetRegion(op.RegionID()))
+		}
+	}
+
+	return m
+}
+
+// DiffMap is a map of StoreDiff.
+type DiffMap map[uint64]*StoreDiff
+
+// GetStoreDiff get storeDiff of specific store.
+func (m DiffMap) GetStoreDiff(id uint64) *StoreDiff {
+	storeDiff, ok := m[id]
+	if !ok {
+		m[id] = &StoreDiff{}
+		storeDiff = m[id]
+	}
+	return storeDiff
+}
+
+// StoreDiff records influences that pending operators will make.
+type StoreDiff struct {
```

Comment:
what does Diff mean here?

Reply:
difference that pending operators influence on stores

Reply:
Maybe use OpInfluence instead of DiffMap, StoreInfluence instead of StoreDiff is better? What do you think?

Reply:
+1